gulpfile.js

```diff
@@ -269,40 +272,46 @@ function resolveInstallerExePath(artifactType) {
   return path.join(zaRoot, artifactName + '-' + pjson.version + artifactType + '-installer.exe');
 }
 
-// download all the installer dependencies so we can package them up into the .exe
-gulp.task('prefetch', function(cb) {
-  let counter=0;
+// prefetch all the installer dependencies so we can package them up into the .exe
+gulp.task('prefetch', function() {
+  let promises = new Set();
+  if (!fs.existsSync(prefetchFolder)) {
+     mkdirp(prefetchFolder);
+  }
   for (var key in reqs) {
     if (reqs.hasOwnProperty(key)) {
       let currentUrl = reqs[key].url;
       let currentKey = key;
       let currentFile = path.join(prefetchFolder, key);
       let alreadyDownloaded = false;
-
+      promises.add(new Promise((resolve,reject)=>{
       // if file is already downloaded, check its sha against the stored one
-      isExistingSHA256Current(currentFile, reqs[currentKey].sha256sum, (alreadyDownloaded)=> {
-        //console.log('For ' + currentFile + ": " + alreadyDownloaded + ", " + reqs[currentKey].bundle);
-        if(!alreadyDownloaded) {
-          // download only what can be included in offline installer
-          if (reqs[currentKey].bundle === 'yes') {
-            counter++;
-            console.log('DOWNLOADING -> ' + currentUrl);
-            request(currentUrl)
-              .pipe(fs.createWriteStream(currentFile)).on('finish',function() {
-                // create a sha256sum file
-                counter--;
-                if(counter===0) {
-                  cb();
+        isExistingSHA256Current(currentFile, reqs[currentKey].sha256sum, (alreadyDownloaded)=> {
+          //console.log('For ' + currentFile + ": " + alreadyDownloaded + ", " + reqs[currentKey].bundle);
+          if(!alreadyDownloaded) {
+            // download only what can be included in offline installer
+            if (reqs[currentKey].bundle === 'yes') {
+              console.log('[INFO] Download ' + currentUrl + ' to ' + currentFile);
+              request(currentUrl).pipe(fs.createWriteStream(currentFile)).on('finish',function(err, res) {
+                if (err) {
+                  console.log(error);
+                  return reject(err);
+                } else {
+                  //console.log('[DEBUG] Downloaded ' + currentFile);
+                  createSHA256File(currentFile);
+                  return resolve(res);
                 }
               });
+            }
           }
-        } else {
-          console.log('DOWNLOADED <- ' + currentFile);
-        }
-      });
+          resolve(true);

```

Comment:
This resolve is getting called right after download is initiated, thus prefetch finishes in a few milliseconds. You need to put it into an else block because of the magic way promises are handled.